src/services/textChanges.ts

```diff
@@ -340,7 +340,9 @@ namespace ts.textChanges {
         }
 
         public insertJsdocCommentBefore(sourceFile: SourceFile, node: HasJSDoc, tag: JSDoc): void {
-            const fnStart = node.getStart(sourceFile);
+            const fnStart = node.kind === SyntaxKind.ArrowFunction

```

Comment:
It should be the caller's responsibility to set the parent node higher in this situation - `textChanges` should always behave as instructed.

Reply:
I hoisted the code to `addJSDocTags`, is it the right place? I added another check for arrow function in class property declaration as well, since I think most programmers would want jsdoc in these two places.